src/extent.c

```diff
@@ -1930,6 +1980,11 @@ extent_merge_impl(tsdn_t *tsdn, arena_t *arena,
 		return true;
 	}
 
+	if (extent_dumpable_get(a) || extent_dumpable_get(b)) {
+		extent_ensure_dumpable(a);
+		extent_ensure_dumpable(b);
```

Comment:
It's not crazy. We didn't want to create yet another source of fragmentation caused by adjacent extents in different states.